reference/performance: add hint for INL_HASH_JOIN and INL_MERGE_JOIN (#1868)
pingcap/docs: reference/performance/optimizer-hints.md

>
> `TIDB_INLJ` is the alias for `INL_JOIN` in TiDB 3.0.x and earlier versions. If you are using any of these versions, you must apply the `TIDB_INLJ(t1_name [, tl_name ...])` syntax for the hint. For the later versions of TiDB, `TIDB_INLJ` and `INL_JOIN` are both valid names for the hint.

### INL_HASH_JOIN

> The `INL_HASH_JOIN(t1_name [, tl_name])` hint tells the optimizer to use the index nested loop hash join algorithm. The conditions for using this algorithm are the same with the conditions for using the index nested loop join algorithm, but in some scenarios the index nested loop hash join algorithm saves more memory resources.

Reviewer: ```suggestion
The `INL_HASH_JOIN(t1_name [, tl_name])` hint tells the optimizer to use the index nested loop hash join algorithm. The conditions for using this algorithm are the same with the conditions for using the index nested loop join algorithm, but in some scenarios the index nested loop hash join algorithm uses less memory.
```

### INL_MERGE_JOIN

> The `INL_MERGE_JOIN(t1_name [, tl_name])` hint tells the optimizer to use the index nested loop merge join algorithm which saves more memory than using `INL_JOIN`. The conditions for using this algorithm include all the conditions for using `INL_JOIN`, plus this one: the column sets of the inner table in join keys is the prefix of the inner table, or the index of the inner table is the prefix of the column sets of the inner table in join keys.

Reviewer: This expression is not a great way. Please try to understand and convey it more directly.
```suggestion
The `INL_MERGE_JOIN(t1_name [, tl_name])` hint tells the optimizer to use the index nested loop merge join algorithm, which consumes less memory than using `INL_JOIN`. The conditions for using this algorithm include all the conditions for using `INL_JOIN` but with one more: the column sets of the inner table in join keys is the prefix of the inner table, or the index of the inner table is the prefix of the column sets of the inner table in join keys.
```
